File: go/examples/basic.go
//go:build ignore
// +build ignore

package main

import (
	"time"

	tracing "github.com/LeDuyViet/quicktrace/go"
)

func main() {
	// Basic usage với default style
	tracer := tracing.NewSimpleTracer("Basic Example")

	tracer.Span("Initialize database")
	time.Sleep(30 * time.Millisecond)

	tracer.Span("Load user data")
	time.Sleep(50 * time.Millisecond)

	tracer.Span("Process data")
	time.Sleep(20 * time.Millisecond)

	tracer.Span("Generate response")
	time.Sleep(10 * time.Millisecond)

	tracer.End()
}

File: js/examples/basic.js
const { Tracer } = require('../tracer.js');

async function main() {
    // Basic usage với default style
    const tracer = new Tracer('Basic Example');

    await new Promise(resolve => setTimeout(resolve, 30));
    tracer.span('Initialize database');

    await new Promise(resolve => setTimeout(resolve, 50));
    tracer.span('Load user data');

    await new Promise(resolve => setTimeout(resolve, 20));
    tracer.span('Process data');

    await new Promise(resolve => setTimeout(resolve, 10));
    tracer.span('Generate response');

    tracer.end();
}

main().catch(console.error); 